### ISSUE TITLE
document patch release process

### ISSUE THREAD

**Issue by skriss** [vmware-tanzu/velero#2472]
Need to update `release-instructions.md` with the steps for a patch release.

**Comment by skriss** [comment]
I'm looking at adding the instructions for a patch release, which involve creating a release branch, cherrypicking over the fixes, adding changelog and docs to both `master` and the release branch, and creating the release tag off of the branch.

I'm hesitant to just mix the unique steps into https://velero.io/docs/v1.4/release-instructions/, as I think that'll get super-complicated. On the other hand, separating it out into its own section means I'll have to duplicate a bunch of the steps that are already there for normal minor-version releases, which increases the maintenance burden of the doc.

What would you all prefer? Intermingle the steps for a patch release in, with a [Patch release only] prefix? Or separate section covering just "Patch release", with some info duplicated?

cc @nrb @carlisia @ashish-amarnath

**Comment by carlisia** [comment]
I vote to not duplicate, at least until we find ourselves hating it then we can split it. I'd go with that first.

**Comment by skriss** [comment]
@nrb would appreciate your input here too since you've recently gone through following the existing instructions.

**Comment by nrb** [comment]
To first document my experience:

Having them mixed was a bit confusing, and required careful reading. It was easy to miss a step because things were intermingled. However, I can fully accept that such a thing is reader error, and that the person doing the release should be reading more carefully.

In terms of what to do right now:

Can some parts be split out, and some parts intermingled? So we have creating branch and cherry picking, which would come first. Changelogs and docs are all the same. That seems to be the most logical for now.

There would be a `[Patch Release Only]` section that says "If a release branch doesn't already exist, make it, and do the cherry-picking of relevant commits". Then the next section would be the existing `(Pre-Release and GA) Changelog and Docs PR`, but maybe renamed to `(Pre-Release, GA, and Patch) Changelog and Docs PR`, which has a note that says "Do all of this in both `master` and the release branch when doing a patch release".

`During Release` step 3 would be modified to say that you should fetch & checkout the release branch.

`During Release` step 4 would be modified to mention that if you're doing a patch release, make the tag from the release branch, not `master`.

For the future:

IMO, this should be automated, as much as possible. These docs are good, but they leave a lot of room for human error. I realize that automation can't make the human decision of what things to cherry-pick, but it's far easier to document "run these 4 scripts" than "do these ~30 steps, with conditionals along the way." I would love to spend time on refining doing this, as it would hopefully mean more people can do the release, and moves us closer to letting CI do the work.

The big hurdles I see to this are:

* How do we amend the changelogs via script? We've got the unreleased contents collection happening, but we have to manually copy/paste those into the canonical changelog files right now.
* How do we verify the site changes automatically? This is probably not really something we can remove a human from. We can get a 👍 /:thumbsdown: on whether it builds, but not on whether things are in the ToC, etc.

Basically, I would love to get the release (for core and plugins) to the point where what we do is hit a button, provide some PRs to cherry pick from for patch releases, and the spot check docs & the image, then do an announcement. But I also realize that's not going to happen in one step, we'll have to iterate.

*puts away soap box* :)

**Comment by nrb** [comment]
@skriss PTAL at #2592 when you get a chance. I think the logic around patch releases will need fleshing out specifically, but it's my attempt at automating the actual release itself.

That PR does _not_ address getting a docs/changelog PR more automated though.
